packages/tokens/build.js

```diff
@@ -249,6 +255,38 @@ StyleDictionary.registerFormat({
   },
 })
 
+/** Formats declarations exports for composite tokens */
+StyleDictionary.registerFormat({
+  name: 'typescript/es6-declarations/composite',
+  formatter: function ({ dictionary, options }) {
+    let tokens = dictionary.allTokens,
+      declaration = '`export declare const ${options.exportName}: {\n'
+
+    if (!options.noSort) {
+      tokens = sortTokensByName(tokens)
+    }
+
+    for (const token of tokens) {
+      let docs = `/** `
+      let valueKeys = '{\n'
+
+      // Generate docs line and object of value types
+      Object.keys(token.value).forEach((key, index) => {
+        docs += `${index !== 0 ? '| ' : ''}${key}: ${token.value[key]} `
+        valueKeys += `    ${key}: ${typeof token.value[key]}\n`
+      })
+
+      docs += `*/\n`
+      declaration += `  ${docs}`
+      declaration += `  ${token.name}: ${valueKeys}`
+      declaration += `  }\n`
+    }
+
+    declaration += '}'
+    return declaration
+  },
+})
+
 /** Creates named type declaration for Themes. Allows for TypeScript autocomplete */
 StyleDictionary.registerFormat({
   name: 'typescript/es6-declarations/theme',
```

Comment:
Just for confirmation, we aren't including Figma export of these tokens because:
- The tooling doesn't support composite tokens OR
- Figma can't meaningfully apply composite tokens similarly to atomic tokens OR
- Some other reason
?

I recall there being a reason, but don't recall what it was or if it'd been documented anywhere.

Reply:
Are you referring to themes or typography?

Reply:
Just confirmed with @jessicawoodin that Figma doesn't accept composite tokens so there's no need to export them